adapters/src/main/kotlin/com/krzykrucz/elesson/currentlesson/lessonprogress/adapters/persistence/LessonProgressDatabaseConnector.kt

```diff
@@ -0,0 +1,24 @@
+package com.krzykrucz.elesson.currentlesson.lessonprogress.adapters.persistence
+
+import arrow.core.toOption
+import arrow.fx.IO
+import com.krzykrucz.elesson.currentlesson.lessonprogress.usecase.CreateLessonProgressView
+import com.krzykrucz.elesson.currentlesson.lessonprogress.usecase.LessonProgress
+import com.krzykrucz.elesson.currentlesson.lessonprogress.usecase.LessonProgressError
+import com.krzykrucz.elesson.currentlesson.monolith.Database
+
+fun createLessonProgressView(): CreateLessonProgressView = { lessonIdentifier ->
```

Comment:
the name is a bit misleading as this function translates `currentLesson` into `CreateLessonProgressView`, maybe it would be better to name it `loadLessgonProgress` ?

Reply:
Done